gameplay/src/Ref.h

```diff
@@ -43,6 +43,41 @@ class Ref
      */
     unsigned int getRefCount() const;
 
+    /**
```

Comment:
Of course it's similar to a smart pointer, but I explicitly simplified it (not including operators *, ->, [], etc) 
I found the existing cases of "free-standing" addref()/release() in the code used for callbacks, and updated them. There were only three. Perhaps there are more places in the code where it calls the user back and SHOULD be using it, but currently it isn't.
If you don't like the class, it's probably easier to merge the commit, then take it out, than to re-create the commit without the class. GitHub isn't as good as git request-pull :-(